### PR TITLE
Initialising project components

### PR DESCRIPTION
**Pull request by Chimwemwe-127001**
# Project components
> In this feature update:
- [x] Initialise React app.
- [x] Add React Redux.
- [x] Structure application files using a `feature folder` approach and use the `ducks pattern` for Redux files.
- [x] Add React Router and set two `<Route>s` and `<Link>s` for the app's navigation

### CODE REVIEW

**Comment by omar-labana** [comment]
Hi @Chimwemwe-127001, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, where the README file isn't updated to show a professional representation of the project, please update it.


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by Chimwemwe-127001** [comment]
Hello @omar-labana. I have updated the README file. Thanks for spotting that.

File: src/components/BooksPage.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            Author:
+            {book.author}
+          </p>
+          <p>
+            Category:
+            {book.Category}
+          </p>
+          <button type="button">Remove</button>
+        </li>
+      ))}
```
Comment (TedLivist) [inline src/components/BooksPage.js]: - [x] Your component for rendering books, in this case, `BookPage.js`, should list the books using a `Book` component. That means you should create a new `Book` component responsible for a single book and extract this mapping functionality into it.
  
  - [x] The new Book component should take in the book data as a prop and render the book title and author (or category) as you have here
  - [x] This new Book component should also have a placeholder Remove button as you have here. The button does not have to work yet.

File: src/components/BooksPage.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      <h1>ADD NEW BOOK</h1>
+      <input type="text" name="Book title" />
+      <select>
+        <option value="">Category</option>
+        <option value="Romance">Romance</option>
+        <option value="Documentary">Documentary</option>
+        <option value="Fiction">Fiction</option>
+        <option value="Crime">Crime</option>
+      </select>
+    </form>
```
Comment (TedLivist) [inline src/components/BooksPage.js]: - [x] In keeping with the SOLID principle, kindly extract this form for a new book into a separate component and render it in this `BookPage.js` component.
- [x] Also add a placeholder **`button`** for adding a new book to your form. This button does not have to work yet.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Chimwemwe-127001,

Good job so far!
✔️ Good documentation
✔️ Correct setup using CRA
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/BooksPage.js (hunk 1 above)
Comment (Chimwemwe-127001) [reply src/components/BooksPage.js]: Done

File: src/components/BooksPage.js (hunk 2 above)
Comment (Chimwemwe-127001) [reply src/components/BooksPage.js]: Done

**Comment by Chimwemwe-127001** [comment]
- Thank you very much @TedLivist for the review. All requested changes have been implemented.
  - Happy coding!! :computer: :smile:

File: src/components/Books.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+Books.propTypes = {
+  bookList: PropTypes.arrayOf(PropTypes.shape(
+    {
+      title: PropTypes.string.isRequired,
+      author: PropTypes.string.isRequired,
+      id: PropTypes.string.isRequired,
+    },
+  )).isRequired,
+};
+
```
Comment (ArthurGC) [inline src/components/Books.js]: - [x] After you implement your changes some issues appears in the console about prop types. It seems an error with the name of the prop `title`.

error-11

File: src/components/BookItem.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      </p>
+      <p>
+        Category:
+        {category}
+      </p>
+      <button type="button" bookItemID={id}>Remove</button>
+    </li>
+  );
+};
+
```
Comment (ArthurGC) [inline src/components/BookItem.js]: - [x] After you implement your changes some issues appears in the console. ♻️ 

error-12

**Review by ArthurGC — CHANGES_REQUESTED** [review]
# Required Changes ♻️ :warning: 
Hi  @Chimwemwe-127001 

Great work on making the changes requested by a previous reviewer. 👏 
You've done well implementing some of the requested changes but there are some issues that you still need to work on to go to the next project but you are almost there! 



## Good Points :+1: :
Great job following most of the first reviewer's suggestions :heavy_check_mark:

## Aspects to improve ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/components/Books.js (hunk 3 above)
Comment (Chimwemwe-127001) [reply src/components/Books.js]: Done

File: src/components/BookItem.js (hunk 4 above)
Comment (Chimwemwe-127001) [reply src/components/BookItem.js]: Done

**Comment by Chimwemwe-127001** [comment]
Thanks for the review @ArthurGC. Hopefully my PR gets approved this time.
Happy coding :computer: :smiley:

**Comment by omar-labana** [comment]
Hi @Chimwemwe-127001, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**